memcached-operator/pkg/controller/memcached/memcached_controller_test.go

```diff
@@ -0,0 +1,83 @@
+package memcached
+
+import (
+	"context"
+	"testing"
+
+	cachev1alpha1 "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis/cache/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// TestMemcachedControllerDeploymentCreate runs ReconcileMemcached.Reconcile()
+// against a fake client that tracks a Memcached object. Reconcile should Get
+// the Memcached resource successfully, Get the namespaced Deployment
+// unsuccessfully, and Create a new Deployment.
+func TestMemcachedControllerDeploymentCreate(t *testing.T) {
+	// Set the logger to development mode for verbose logs.
+	logf.SetLogger(logf.ZapLogger(true))
+
+	var (
+		name            = "memcached-operator"
+		namespace       = "memcached"
+		replicas  int32 = 3
+	)
+
+	// A Memcached resource with metadata and spec.
+	memcached := &cachev1alpha1.Memcached{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: cachev1alpha1.MemcachedSpec{
+			Size: replicas, // Set desired number of Memcached replicas.
+		},
+	}
+	// Objects to track in the fake client.
+	objs := []runtime.Object{
+		memcached,
+	}
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(cachev1alpha1.SchemeGroupVersion, memcached)
+	// Create a fake client to mock API calls.
+	cl := fake.NewFakeClient(objs...)
+	// Create a ReconcileMemcached object with the scheme and fake client.
+	r := &ReconcileMemcached{client: cl, scheme: s}
+
+	// Mock request to simulate Reconcile() being called on an event for a
+	// watched resource .
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	res, err := r.Reconcile(req)
+	if err != nil {
+		t.Fatalf("reconcile: (%v)", err)
+	}
+	// Check the result of reconciliation to make sure it has the desired state.
+	if !res.Requeue {
+		t.Error("reconcile did not requeue request as expected")
+	}
+
+	// Check if deployment has been created and has the correct size.
+	dep := &appsv1.Deployment{}
+	err = cl.Get(context.TODO(), req.NamespacedName, dep)
+	if err != nil {
+		t.Fatalf("get deployment: (%v)", err)
+	}
+	dsize := *dep.Spec.Replicas
+	if dsize != replicas {
+		t.Errorf("dep size (%d) is not the expected size (%d)", dsize, replicas)
+	}
```

Comment:
Was just about to comment on checking for the deployment before you updated it 😄 

Can we also check if the reconciler successfully updates the CR's `status.nodes` with the pod names?
Of course we'll first have to manually create the 3 sample pods ourselves with the fake client.